src/components/Wallet/DirectExtendedPublicKeyImporter.jsx

```diff
@@ -98,25 +95,27 @@ class HardwareWalletExtendedPublicKeyImporter extends React.Component {
         </Box>
         <InteractionMessages
           messages={interaction.messagesFor({ state: status })}
-          excludeCodes={["bip32"]}
         />
-        <FormHelperText className="text-danger">
-          {extendedPublicKeyError}
-        </FormHelperText>
+        <FormHelperText error>{extendedPublicKeyError}</FormHelperText>
       </Box>
     );
   };
 
   import = async () => {
     const {
       validateAndSetExtendedPublicKey,
+      validateAndSetRootFingerprint,
       enableChangeMethod,
       disableChangeMethod,
     } = this.props;
     disableChangeMethod();
     this.setState({ extendedPublicKeyError: "", status: ACTIVE });
     try {
-      const extendedPublicKey = await this.interaction().run();
+      const result = await this.interaction().run();
+      const extendedPublicKey = result.xpub;
+      validateAndSetRootFingerprint(result.rootFingerprint, (error) => {
+        this.setState({ extendedPublicKeyError: error, status: PENDING });
```

Comment:
what happens if the next validation also gives an error? It will overwrite this one right? Is this desired? (and actually since we can't guarantee the order of how these are fired then we can't even know which will overwrite which.